addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComSerialConnector.java

```diff
@@ -0,0 +1,245 @@
+		_listeners.remove(listener);
+	}
+
+	public class SerialReader extends Thread implements SerialPortEventListener {
+		boolean interrupted = false;
+		InputStream in;
+
+		public SerialReader(InputStream in) {
+			this.in = in;
+		}
+		
+		@Override
+		public void interrupt() {
+			interrupted = true;
+			super.interrupt();
+		    try {
+		      in.close();
+		    } catch (IOException e) {} // quietly close
+		}
+
+		public void run() {
+			final int dataBufferMaxLen = Byte.MAX_VALUE;
+
+			byte[] dataBuffer = new byte[dataBufferMaxLen];
+
+			int msgLen = 0;
+			int index = 0;
+			boolean start_found = false;
+
+			logger.debug("Data listener started");
+			
+			// RXTX serial port library causes high CPU load
+			// Start event listener, which will just sleep and slow down event loop
+			try {
+				serialPort.addEventListener(this);
+				serialPort.notifyOnDataAvailable(true);
+			} catch (TooManyListenersException e) {
+			}
+			
+			try {
+
+				byte[] tmpData = new byte[20];
+				int len = -1;
+
+				while ((len = in.read(tmpData)) > 0 && interrupted != true) {
+					
+					byte[] logData = Arrays.copyOf(tmpData, len);
+					logger.trace("Received data (len={}): {}",
+							len,
+							DatatypeConverter.printHexBinary(logData));
+					
+					for (int i = 0; i < len; i++) {
+
+						if (index > dataBufferMaxLen) {
+							// too many bytes received, try to find new start
+							start_found = false;
+						}
+
+						if (start_found == false && tmpData[i] > 0) {
+
+							start_found = true;
+							index = 0;
+							dataBuffer[index++] = tmpData[i];
+							msgLen = tmpData[i] + 1;
+
+						} else if (start_found) {
+
+							dataBuffer[index++] = tmpData[i];
+
+							if (index == msgLen) {
+
+								// whole message received, send an event
+
+								byte[] msg = new byte[msgLen];
+
+								for (int j = 0; j < msgLen; j++)
+									msg[j] = dataBuffer[j];
+
+								RFXComMessageReceivedEvent event = new RFXComMessageReceivedEvent(
+										this);
+
+								try {
+									Iterator<RFXComEventListener> iterator = _listeners
+											.iterator();
+
+									while (iterator.hasNext()) {
+										((RFXComEventListener) iterator.next())
+												.packetReceived(event, msg);
+									}
+
+								} catch (Exception e) {
+									logger.error("Event listener invoking error", e);
+								}
+
+								// find new start
+								start_found = false;
+							}
+						}
+					}
+				}
+			} catch (InterruptedIOException e) {
+			      Thread.currentThread().interrupt();
+			      logger.error("Interrupted via InterruptedIOException");
+			} catch (IOException e) {
+				logger.error("Reading from serial port failed", e);
+			}
+			
+			serialPort.removeEventListener();
+			logger.debug("Data listener stopped");
+		}
+
+		@Override
+		public void serialEvent(SerialPortEvent arg0) {
+			try {
+				logger.trace("RXTX library CPU load workaround, sleep forever");
+				sleep(Long.MAX_VALUE);
```

Comment:
This looks very fishy, could you add a comment with a reference to an issue, why this sleep is a good idea?